### PR TITLE
Add AttribSysVault Editor

### PR DESCRIPTION
**Pull request by derneuere**
This pull request enables the user to change the collection hash of a given resource. It will make it possible to add a new car without using a hex editor.

This was done by enabling saving of AttribSys. I changed the way "PtrN" are saved, as it only saved some pointers and I didn't find any explanation of what was trying to be achieved. The rest is unchanged.

I also added the option to view any entry with the debug menu.

I tested it and it worked fine, but some testing would be great! This is the first time I am saving BinaryStreams 😄

### CODE REVIEW

**Comment by burninrubber0** [comment]
Works in my testing, but the new hash needs to be byteswapped so it's little endian.

The debug menu works well for AttribSysVault resources but on unimplemented types (e.g., StreamedDeformation) it exposes fields for what I assume is the parent object. Best to disable it for all but the implemented types if possible.

Also, unless you're planning to add different primary functionality, it'd be worth using double click to edit the collection hash in addition to the context menu.

**Comment by burninrubber0** [comment]
Works in my testing, but the new hash needs to be byteswapped so it's little endian.

The debug menu works well for AttribSysVault resources but on unimplemented types (e.g., StreamedDeformation) it exposes fields for what I assume is the parent object. Best to disable it for all but the implemented types if possible.

Also, unless you're planning to add different primary functionality, it'd be worth using double click to edit the collection hash in addition to the context menu.

File: VaultFormat/AttribSysVaultForm.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                return;
+
+            for (int i = 0; i < AttribSys.DataChunks.Count; i++)
+            {
+                DataChunk chunk = AttribSys.DataChunks[i];
+
+                string[] value = {
+                    chunk.ClassName,
+                    chunk.ClassHash.ToString("X8"),
+                    chunk.CollectionHash.ToString("X8"),
```
Comment (burninrubber0) [inline VaultFormat/AttribSysVaultForm.cs:50]:
```suggestion
                    chunk.ClassHash.ToString("X16"),
                    chunk.CollectionHash.ToString("X16"),
```

File: VaultFormat/AttribSysVaultForm.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            public void Swap()
+            {
+                this.Direction = !this.Direction;
+            }
+        }
+
+        private void lstVehicles_SelectedIndexChanged(object sender, EventArgs e)
+        {
+
+        }
```
Comment (burninrubber0) [inline VaultFormat/AttribSysVaultForm.cs:118]: Is this needed for something?

**Comment by derneuere** [comment]
I plan to add editing of data chunks / attributes. It would make sense to open up the editor with a double click :)

**Comment by burninrubber0** [comment]
Looks ready for merge after that last one is resolved.

**Comment by derneuere** [comment]
Done! 😄

**Review by burninrubber0 — APPROVED** [review]
All good now, looking forward to seeing where this goes!

**Pull request merged**